scripts/react_native_pods.rb

```diff
@@ -680,19 +683,19 @@ def downloadAndConfigureHermesSource(react_native_path)
 
   hermes_tarball_url = hermes_tarball_base_url + hermes_tag
   hermes_tag_sha = %x[git ls-remote https://github.com/facebook/hermes #{hermes_tag} | cut -f 1].strip
-  hermes_tarball_path = "#{download_dir}/hermes-#{hermes_tag_sha}.tar.gz"
+  hermes_tarball_path = download_dir.join("hermes-#{hermes_tag_sha}.tar.gz")
 
   if (!File.exist?(hermes_tarball_path))
     Pod::UI.puts "[Hermes] Downloading Hermes source code..."
-    system("curl #{hermes_tarball_url} -Lo #{hermes_tarball_path}")
+    system("curl #{hermes_tarball_url} -Lo #{hermes_tarball_path.to_s}")
   end
   Pod::UI.puts "[Hermes] Extracting Hermes tarball (#{hermes_tag_sha.slice(0,6)})"
-  system("tar -zxf #{hermes_tarball_path} --strip-components=1 --directory #{hermes_dir}")
+  system("tar -zxf #{hermes_tarball_path.to_s} --strip-components=1 --directory #{hermes_dir.to_s}")
 
   # TODO: Integrate this temporary hermes-engine.podspec into the actual one located in facebook/hermes
-  system("cp #{sdks_dir}/hermes-engine.podspec #{hermes_dir}/hermes-engine.podspec")
+  FileUtils.cp(sdks_dir.join("hermes-engine.podspec"), hermes_dir.join("hermes-engine.podspec"))
```

Comment:
I will make a separate PR targeting `main` since this section has changed a bit since the cut.